scripts/tybalt_vae.py

```diff
@@ -0,0 +1,320 @@
+rnaseq_reconstruct = decoder_to_reconstruct(z)
+
+
+# ## Connect the encoder and decoder to make the VAE
+# 
+# The `CustomVariationalLayer()` includes the VAE loss function (reconstruction + (beta * KL)), which is what will drive our model to learn an interpretable representation of gene expression space.
+# 
+# The VAE is compiled with an Adam optimizer and built-in custom loss function. The `loss_weights` parameter ensures beta is updated at each epoch end callback
+
+# In[12]:
+
+adam = optimizers.Adam(lr=learning_rate)
+vae_layer = CustomVariationalLayer()([rnaseq_input, rnaseq_reconstruct])
+vae = Model(rnaseq_input, vae_layer)
+vae.compile(optimizer=adam, loss=None, loss_weights=[beta])
+
+vae.summary()
+
+
+# In[13]:
+
+# Visualize the connections of the custom VAE model
+output_model_file = os.path.join('figures', 'onehidden_vae_architecture.png')
+plot_model(vae, to_file=output_model_file)
+
+SVG(model_to_dot(vae).create(prog='dot', format='svg'))
+
+
+# ## Train the model
+# 
+# The training data is shuffled after every epoch and 10% of the data is heldout for calculating validation loss.
+
+# In[14]:
+
+get_ipython().run_cell_magic('time', '', 'hist = vae.fit(np.array(rnaseq_train_df),\n               shuffle=True,\n               epochs=epochs,\n               batch_size=batch_size,\n               validation_data=(np.array(rnaseq_test_df), np.array(rnaseq_test_df)),\n               callbacks=[WarmUpCallback(beta, kappa)])')
+
+
+# In[15]:
+
+# Visualize training performance
+history_df = pd.DataFrame(hist.history)
+hist_plot_file = os.path.join('figures', 'onehidden_vae_training.pdf')
+ax = history_df.plot()
+ax.set_xlabel('Epochs')
+ax.set_ylabel('VAE Loss')
+fig = ax.get_figure()
+fig.savefig(hist_plot_file)
+
+
+# ## Compile and output trained models
+# 
+# We are interested in:
+# 
+# 1. The model to encode/compress the input gene expression data
+#   * Can be possibly used to compress other tumors
+# 2. The model to decode/decompress the latent space back into gene expression space
+#   * This is our generative model
+# 3. The latent space compression of all pan cancer TCGA samples
+#   * Non-linear reduced dimension representation of tumors can be used as features for various tasks
+#     * Supervised learning tasks predicting specific gene inactivation events
+#     * Interpolating across this space to observe how gene expression changes between two cancer states
+# 4. The weights used to compress each latent node
+#   * Potentially indicate learned biology differentially activating tumors
+
+# ### Encoder model
+
+# In[16]:
+
+# Model to compress input
+encoder = Model(rnaseq_input, z_mean_encoded)
+
+
+# In[17]:
+
+# Encode rnaseq into the hidden/latent representation - and save output
+encoded_rnaseq_df = encoder.predict_on_batch(rnaseq_df)
+encoded_rnaseq_df = pd.DataFrame(encoded_rnaseq_df, index=rnaseq_df.index)
+
+encoded_rnaseq_df.columns.name = 'sample_id'
+encoded_rnaseq_df.columns = encoded_rnaseq_df.columns + 1
```

Comment:
This is simply converting `range(0,100)` to `range(1,101)` in the column names. So the first column of the data frame is `latent feature 1` instead of `latent feature 0`. This is done to maintain consistent naming convention for features